src/patch/createPatch.ts

```diff
@@ -41,8 +41,14 @@ function _diff(from: any, to: any, startingPath?: JsonPointer): Operation[] {
 	return operations;
 }
 
-export function diff<T, U>(from: T, to: U): Patch<T, U> {
-	return <Patch<T, U>> createPatch(_diff(from, to));
+export function diff<T>(to: T): Patch<any, T>;
+export function diff<T, U>(from: T, to: U): Patch<T, U>;
+export function diff(from: any, to?: any) {
+	if (!to) {
+		to = from;
+		from = {};
+	}
+	return createPatch(_diff(from, to));
```

Comment:
In my experience the `from` usually comes first. I'm drawing from a limited sample, git and a couple diff libraries, but I would find `diff(to, from)` confusing. Would having a separate method for the single argument case help out the confusion?